frontend/src/components/utils/autoRefreshHandler.tsx

```diff
@@ -49,48 +50,64 @@ export default (
           items: Array<any>(),
           itemsLength: config.count,
           display: false,
-          hasMore: true
+          hasMore: true,
+          loading: false
         });
       }
     }
 
+    regularFetchInfo = () => {
+      if (config.dashboard) {
+        this.setState({ hasMore: false });
+      }
+      if (this.state.itemsLength > 0) {
+        this.fetchInfo(this.state.itemsLength);
+        if (this.timer !== null) {
+          this.timer = setTimeout(this.regularFetchInfo, 10000);
+        }
+      }
+    };
+
     fetchInfo = (count: number) => {
       config
         .fetchDataFn(count)
         .then((items: any) => {
           if (items.length > 0) {
-            this.setState({ items, display: true });
+            this.setState({ items, itemsLength: items.length, display: true });
+            if (items.length < config.count) {
+              this.setState({ hasMore: false });
+            }
           } else {
-            this.setState({ hasMore: false, display: true });
+            this.setState({ hasMore: false, display: true, itemsLength: 0 });
           }
         })
-        .catch((err: any) => {
+        .catch((err: Error) => {
           console.error(err);
         });
     };
 
-    regularFetchInfo = () => {
-      this.fetchInfo(this.state.itemsLength);
-      if (this.timer !== null) {
-        this.timer = setTimeout(this.regularFetchInfo, 10000);
-      }
-    };
-
     fetchMoreData = async () => {
-      if (config.dashboard) {
-        this.setState({ hasMore: false });
-        return;
-      }
-      if (this.state.items.length > 0) {
+      this.setState({ loading: true });
```

Comment:
This may set the loading = true, and then never go into the `if` statement, and thus never call the fetching data method, and ever resolve, so loading will never get to false.